effects/letter-clock/letter-clock.c

```diff
@@ -0,0 +1,121 @@
+#include "effect.h"
+
+/* Search following header files for useful procedures. */
+#include "blitter.h" /* blitter handling routines */
+#include "copper.h"  /* copper list construction */
+#include "bitmap.h"  /* bitmap structure */
+#include "palette.h" /* palette structure */
+#include "circle.h"
+
+#define WIDTH 320
+#define HEIGHT 256
+#define DEPTH 5
+#define CENTER_X (WIDTH / 2)
+#define CENTER_Y (HEIGHT / 2 + 25)
+#define SMALL_R 20
+#define FIRST_LENS 3
+#define SECOND_LENS 4
+
+#include "data/clock-bg.c"
+#include "data/clock-palette.c"
+#include "data/movement_data.c"
+
+static CopListT *cp;
+static BitmapT *bckg;
+int old_x[2], old_y[2];
+
+// Only one lens is moved at a time. The circular movement is partially preprocessed.
+static void Circles(int intra_x, int intra_y, int progress, int smallR)
+{
+    int by_two = progress & 1;
+    int plane = by_two ? FIRST_LENS : SECOND_LENS;
+
+    int x = circle_movements[by_two].x_cos[progress] + intra_x;
```

Comment:
Is it really necessary to provide code that generates `circle_movements` array? After all you could use `SIN` and `COS` functions from `fx.h` file. Why make things more complex than they need to be?